templates/common/_base/units/machine-config-daemon-pull.service.yaml

```diff
@@ -0,0 +1,43 @@
+name: "machine-config-daemon-pull.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Machine Config Daemon Pull
+  # Make sure it runs only on OSTree booted system
+  ConditionPathExists=/run/ostree-booted
+  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+  # We only want to run on 4.3 clusters and above; this came from
+  # https://github.com/coreos/coreos-assembler/pull/768
+  ConditionPathExists=/sysroot/.coreos-aleph-version.json
+  # Skip this unit if MCD binary already exists at libexec
+  ConditionPathExists=!/usr/libexec/bin/machine-config-daemon
+  After=ignition-firstboot-complete.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  # Extract machine-config-daemon binary from MCD container
+  ExecStart=/usr/bin/sh -c "while ! /usr/bin/podman pull --authfile=/var/lib/kubelet/config.json '{{ .Images.setupEtcdEnvKey }}'; do sleep 5; done"
+  ExecStart=/usr/bin/sh -c "/usr/bin/podman run --authfile=/var/lib/kubelet/config.json --quiet --net=host --entrypoint=cat '{{ .Images.setupEtcdEnvKey }}' /usr/bin/machine-config-daemon > /usr/local/bin/machine-config-daemon"
+  ExecStart=/usr/bin/chmod +x /usr/local/bin/machine-config-daemon
```

Comment:
I was going to say to write this in `/run` instead per https://github.com/coreos/fedora-coreos-tracker/issues/354# but...then I remembered we still need this for `machine-config-daemon-host.service`.  Ugg.

Ideally at some point we get rid of `machine-config-daemon-host.service` - I think that only exists because of `spc_t` issues but https://bugzilla.redhat.com/show_bug.cgi?id=1839065 should let us fix that.

Alternatively, we could also dynamically inject the binary into the host from the *pod* case too (i.e. not firstboot).  That would make everything symmetric.

The concern here is "what updates that binary" - the semantic here is we only write it on firstboot, which is going to mean it becomes stale.

So if we dynamically inject it in both cases, ownership and lifecycle is clear.

Reply:
Indeed it will be more consistent if we pull m-c-d binary from image during subsequent upgrades as well. This also mean that we will no longer need to maintain m-c-d rpm package.